wordpress-mobile/WordPress-Android
# Fix memory leaks SiteCreation Segments and Domains screens

I've completely forgotten about these changes @zwarm and I implemented during a pair programming session a couple of weeks ago.

Fixes leaks in Segments and Domains Site Creation's screens. (PreviewFragment is tbd)

Fragment can live even after its views were destroyed (eg on backstack). If the fragment keeps references to views after `onViewDestroyed` is invoked, all the views are leaked. This PR replaces class fields with kotlin synthetic accessors.
One extra change is, that the setupContent method is invoked from within "onViewCreated". It's the correct place anyway, since "onCreateView" should be taking care of view creation but not initialisation.

To test:
1. My Site
2. Switch Site
3. Turn on airplane mode
4. Click on "+" and "Create WordPress.com site"
5. Observe the error screen is shown
6. Click on retry and observe a progress is shown for a bit and error screen is shown again
7. Turn off airplane mode and click on retry
8. Segments screen is shown
9. Select a segment
10. Turn on airplane mode and type something
11. Error is shown
12. Turn off airplane mode
13. Click on retry
14. Observe results are shown
15. Select a domain and click on Create Site button
PR submission checklist:
